### PR TITLE
Move 'imported docs' list into sidebar

### PR DESCRIPTION
**Pull request by ChrisBAshton**
We'd already established in #TBC that showing a list of all apps in the
sidebar was not very useful, and temporarily removed everything
but the search form from the sidebar.

The traditional layout parses the page contents and surfaces
anchor links in a 'table of contents' style. Something similar to
that - and useful, and contextual - is to surface all the imported
documents in the sidebar. We can therefore remove them from the main
repo overview page.

Before:

before

After:

after

Trello: https://trello.com/c/oUsgzZnk/247-standardise-how-we-represent-repositories-in-the-dev-docs

### CODE REVIEW

File: source/layouts/repo_layout.html.erb [hunk 1]
```diff
@@ -1,26 +1,37 @@
...
+          end
+          items << GovukPublishingComponents.render("govuk_publishing_components/components/list", { visible_counters: true, items: subdir_items })
+        end
+      end
+
+      imported_docs = GitHubRepoFetcher.instance.docs(repo.repo_name)
+      unless imported_docs.nil? || imported_docs.empty?
+        add_imported_docs.call(imported_docs)
+      end
+    end
```
Comment (ChrisBAshton) [inline source/layouts/repo_layout.html.erb:32]: ☝️  this all looks painful but that's mostly down to the lack of syntax highlighting! This is basically a lift and shift of what was already in `source/partials/repo/_imported_docs.html.erb` (now deleted), converted to pure Ruby (rather than ERB) as we're now using a component to render the data.

**Review by benthorner — COMMENTED** [review]
This change makes a lot of sense ➕ 👍.

I have two more major concerns I'd like to look at before this goes in.

1. The structure of the new sidebar is unclear:

  - It's unclear how `<app>` differs from `README` below it; the latter is really an anchor in the former but has the same heading level and also behaves differently to all the other links. One option would be to just drop the `README` item.
 
  - It's unclear what the other items are, for someone not familiar with the repo and the fact that we import files in `docs/`. Having some more structure - like a heading - would help. One of my comments has a suggestion about that 🤞.

2. The embedded Ruby code is hard to follow.

- The previous partial was manageable but the new code is more complex - I struggled. Even if we just dump all of it in a Ruby file as-is, that will still be a big improvement as at least then we have some syntax highlighting to help.

All that said, I think this is still a net improvement over what we have now, so I'm open to merging and reviewing changes in follow-up PRs.

File: source/layouts/repo_layout.html.erb [hunk 2]
```diff
@@ -1,26 +1,37 @@
...
-        <% end %>
-        </ul>
-      </li>
-    <% end %>
-  </ul>
+    unless repo.private_repo?
+      items << sanitize("<a href=\"/repos/#{repo.repo_name}.html#readme\" rel=\"noopener\" class=\"govuk-link\">README</a>")
+
+      add_imported_docs = -> (docs_to_import) do
+        docs_subdirs = docs_to_import.map { |doc| doc[:path].split("/")[3..(doc[:path].split("/").count - 2)].join("/") }.uniq.reject(&:empty?)
```
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply source/layouts/repo_layout.html.erb]: It's now been simplified and split into HTML and Ruby blocks of code, so syntax highlighting is back 🎉 Hopefully that resolves this comment.

File: source/layouts/repo_layout.html.erb [hunk 3]
```diff
@@ -1,26 +1,37 @@
 <% content_for :sidebar do %>
-  <ul>
-    <li>
-      <a href="/apps/by-team.html">
-        <span>Repository ownership by team</span>
-      </a>
-    </li>
+  <%
+    items = [
+      sanitize("<a href=\"/repos/#{repo.repo_name}.html\" rel=\"noopener\" class=\"govuk-link\">#{repo.repo_name}</a>"),
```
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply source/layouts/repo_layout.html.erb]: Have added something to the commit.

File: source/stylesheets/screen.css.scss [hunk 4]
```diff
@@ -78,3 +78,8 @@
 .technical-documentation > p:first-child {
   margin: govuk-spacing(3) 0 0 0;
 }
+
+/* high specificity required, otherwise overwritten by `.toc ul` */
+.gem-c-list.govuk-list.govuk-list--bullet {
+  @extend %govuk-list--bullet;
```
[the remark this answers is not in the document]
Comment (ChrisBAshton) [reply source/stylesheets/screen.css.scss]: Agreed, I've switched to native HTML and a tiny bit of custom CSS.

**Review by KludgeKML — APPROVED** [review]
Seems reasonable!
